### DISCUSSION TITLE
[Question] How to set a constant velocity for a RigidObject?

### DISCUSSION THREAD

**Discussion by JPBG-USP** [isaac-sim/IsaacLab#1412] [Q&A] [answered]
I'm working on a project where I would need a dynamic obstacle. My idea is to create a RigidObject (a cube) and set it to a constant speed. 

What I came across is that when using `mdp.reset_root_state_uniform` , or `mdp.push_by_setting_velocity`, the speed is only applied one instant of time dt.

I would like to know if there is any way to define a constant speed during the environment time. The speed of the cube is defined by an event class.

**Comment by StrainFlow** [comment]
If it's sliding on the ground you should be able to give it a material with a friction coefficient of 0 and then give it an initial velocity. This deformable body shows how to set the initial state of a prim. (it's a deformable prim, but the init state should be similar for a rigid body). 

https://isaac-sim.github.io/IsaacLab/main/source/tutorials/01_assets/run_deformable_object.html

**Comment by JPBG-USP** [comment, marked as answer]
**Feedback and Cube Configuration**
Here is the configuration that worked:
```python
cube = RigidObjectCfg(
    prim_path="{ENV_REGEX_NS}/Cube",
    spawn=sim_utils.CuboidCfg(
        size=(0.5, 0.5, CUBE_HEIGHT),
        rigid_props=sim_utils.RigidBodyPropertiesCfg(),
        mass_props=sim_utils.MassPropertiesCfg(mass=70.0),
        collision_props=sim_utils.CollisionPropertiesCfg(collision_enabled=True),
        visual_material=sim_utils.PreviewSurfaceCfg(
            diffuse_color=(0.21, 0.5, 0.73),
            metallic=0.2
        ),
        physics_material=sim_utils.RigidBodyMaterialCfg(
            static_friction=0.001,
            dynamic_friction=0.001,
            improve_patch_friction=False,
            friction_combine_mode='min'
        )
    ),
    init_state=RigidObjectCfg.InitialStateCfg(
        pos=(4.0, 0.0, CUBE_HEIGHT / 2),
        lin_vel=(-0.5, 0.0, 0.0)
    )
)

```
**Note:**
I encountered an issue where the cube wouldn't slide on the ground, even with friction values set to zero. The solution was to set friction_combine_mode to 'min'. This way, during the interaction between the cube and the ground, the lowest friction value is applied (in this case, 0).

**Reply by xiasun** [reply]
Any idea how to set velocity at runtime? When I tried to do this it always raises an error
```
# create handles for forklift
SingleGeometryPrim(f"/World/Forklift", collision=False)
forklift_obj = SingleRigidPrim(f"/World/Forklift", mass=2500)
forklift_obj.set_linear_velocity(torch.tensor([-1, 1, 0]))
```
Error message
```
PhysX error: PxRigidDynamic::setLinearVelocity(): it is illegal to call this method if PxSceneFlag::eENABLE_DIRECT_GPU_API is enabled!
```
Seems this sort of APIs can not be used at runtime.

Many thanks!
